modules/apps/site-navigation/site-navigation-language-web/src/main/java/com/liferay/site/navigation/language/web/internal/portal/settings/configuration/admin/display/LocaleFriendlyUrlPortalSettingsConfigurationScreenContributor.java

```diff
@@ -0,0 +1,132 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.site.navigation.language.web.internal.portal.settings.configuration.admin.display;
+
+import com.liferay.petra.reflect.ReflectionUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.security.auth.CompanyThreadLocal;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.PropsKeys;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.settings.configuration.admin.display.PortalSettingsConfigurationScreenContributor;
+import com.liferay.portal.util.PropsUtil;
+import com.liferay.portal.util.PropsValues;
+import com.liferay.site.navigation.language.web.internal.configuration.SiteNavigationLocalePrependFriendlyUrlStyleConfiguration;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+/**
+ * @author Albert Gomes Cabral
+ */
+@Component(service = PortalSettingsConfigurationScreenContributor.class)
+public class LocaleFriendlyUrlPortalSettingsConfigurationScreenContributor
+	implements PortalSettingsConfigurationScreenContributor {
+
+	@Override
+	public String getCategoryKey() {
+		return "localization";
+	}
+
+	@Override
+	public String getJspPath() {
+		return "/portal_settings/locale_friendly_url_style.jsp";
+	}
+
+	@Override
+	public String getKey() {
+		return "locale-friendly-url-configuration";
+	}
+
+	@Override
+	public String getName(Locale locale) {
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", locale, getClass());
+
+		return _language.get(
+			resourceBundle,
+			"site-navigation-locale-prepend-friendly-url-configuration-name");
+	}
+
+	@Override
+	public String getSaveMVCActionCommandName() {
+		return "/configuration_admin/save_locale_friendly_url_configuration";
+	}
+
+	@Override
+	public ServletContext getServletContext() {
+		return _servletContext;
+	}
+
+	@Override
+	public void setAttributes(
+		HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse) {
+
+		SiteNavigationLocalePrependFriendlyUrlStyleConfiguration
+			siteNavigationLocaleFriendlyUrlConfiguration = null;
+
+		try {
+			siteNavigationLocaleFriendlyUrlConfiguration =
+				_configurationProvider.getCompanyConfiguration(
+					SiteNavigationLocalePrependFriendlyUrlStyleConfiguration.
+						class,
+					CompanyThreadLocal.getCompanyId());
```

Comment:
@alberttgomes, please use the PortalPreferences instead to keep the information in a single database table